src/targets/services/cleanOrphanAccounts.js

```diff
@@ -0,0 +1,81 @@
+const cozyFetch = require('../../lib/services/cozyFetch')
+
+const logMessagePrefix = level =>
+  `[cozy-collect][cleanOrphanAccounts][${level}]:`
+
+const output = level =>
+  function() {
+    console.log.apply(
+      console,
+      [logMessagePrefix(level)].concat(
+        // Node < 7 does not know about Object.values
+        Object.keys(arguments).map(key => arguments[key])
+      )
+    )
+  }
+
+const log = {
+  error: output('error'),
+  info: output('info')
+}
+
+const handleFetchError = error => log.error(error.message)
+
+log.info('Fetching accounts...')
+
+cozyFetch(
+  'GET',
+  '/data/io.cozy.accounts/_all_docs?include_docs=true',
+  null,
+  true
+)
+  .then(accounts => {
+    log.info(`${accounts.length || 0} account(s) has been found.`)
+    log.info('Fetching konnector triggers...')
+
+    cozyFetch('GET', '/jobs/triggers?Worker=konnector', null, true)
+      .then(result => {
+        const hasTriggers = result && result.data && result.data.length
+        const triggers = result.data
+
+        if (hasTriggers) {
+          log.info(
+            `${triggers.length || 0} konnector trigger(s) has been found.`
+          )
+        }
+
+        const orphanAccounts = hasTriggers
+          ? accounts.filter(
+              account =>
+                !triggers.find(
+                  trigger =>
+                    trigger.attributes &&
+                    trigger.attributes.message &&
+                    trigger.attributes.message.account &&
+                    trigger.attributes.message.account === account._id
+                )
+            )
+          : accounts
+
+        log.info(
+          `${orphanAccounts.length || 0} orphan account(s) has been found.`
+        )
+
+        if (orphanAccounts.length) {
+          orphanAccounts.forEach(account => {
+            log.info(`Deleting account ${account._id}...`)
+
+            cozyFetch(
+              'DELETE',
+              `/data/io.cozy.accounts/${account._id}?rev=${account._rev}`,
+              null,
+              true
+            )
```

Comment:
Would be more efficient to do a bulk delete here.

1. Map on the documents on add `_deleted: true`.
2. Use the _bulk_docs route

Exemple ici : https://gitlab.cozycloud.cc/labs/ACH/blob/master/scripts/deleteOrphanOperations.js#L22

Reply:
Very good point, thanks !